srcts/src/components/_utils.ts

```diff
@@ -50,10 +50,26 @@ function doWindowResizeOnElementResize(el: HTMLElement): void {
   $(el).data("window-resize-observer", ro);
 }
 
+function getAllFocusableChildren(el: HTMLElement): HTMLElement[] {
+  const selectors = [
+    "[href]",
+    "input:not([disabled])",
+    "button:not([disabled])",
+    "select:not([disabled])",
+    "summary:not(:disabled)",
+    "details:not([disabled])",
+    "textarea:not([disabled])",
+    '[tabindex]:not([tabindex="-1"]):not([disabled])',
```

Comment:
By any chance do you have a link to a resource where I can learn more about this?

Reply:
I can't find exactly where I found this list (I should have copied a link!) but here's a similar list we could use for reference: https://github.com/c-vetter/get-focusables/blob/master/src/selectors.js

Reply:
In general, https://github.com/rstudio/bslib/pull/557/commits/44cbe67d7da51d2bca71785acb0bb0d3fefa5964 gives me the heebie-jeebies. Can we design the full-screen markup in such a way that the close button is focusable without needing to takeover control of focus handling?

Reply:
Yeah, we could use a `<dialog>` modal, but I think all native alternatives that don't require us to manage tab focus will require us to do a decent bit of DOM manipulation to move things around.

The big thing that commit is trying to solve is that, in addition to making the close button focusable, we need to also trap tab focus in the full screen area.

Reply:
Ok, I'm coming around to the approach you're using here. Let's make sure to add a comment about this and perhaps a link to https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/dialog_role#required_javascript_features  where we call `getAllFocusableChildren()`

Reply:
And also, according to [this](https://stackoverflow.com/a/30753870/1583084), we should also be excluding `tabindex=-1`...

Reply:
Agreed, I think we should start with that linked list and then verify that it's up to date. (I'll do this now.)